src/expressions/loop-expr.md

```diff
@@ -123,11 +123,32 @@ while let Some(v @ 1) | Some(v @ 2) = vals.pop() {
 }
 ```
 
+The expression cannot be a [lazy boolean operator expression][_LazyBooleanOperatorExpression_].
+Use of a lazy boolean operator is ambiguous with a planned feature change
+of the language (the implementation of if-let chains - see [eRFC 2947][_eRFCIfLetChain_]).
+When lazy boolean operator expression is desired, this can be achieved
+by using parenthesis as below:
+
+<!-- ignore: psuedo code -->
+```rust,ignore
+// Before...
+while let PAT = EXPR && EXPR { .. }
+
+// After...
+while let PAT = ( EXPR && EXPR ) { .. }
+
+// Before...
+while let PAT = EXPR || EXPR { .. }
+
+// After...
+while let PAT = ( EXPR || EXPR ) { .. }
+```
```

Comment:
This is user advice and discussion of unstable features, which is off-topic for the reference.

Reply:
Then perhaps it should also be removed from the page about `if` expressions?

Reply:
Probably yes, cc @ehuss

Reply:
I think it's fine to keep this information, though place it in a "note" block. I don't see a reason to *never* talk about such things, since it is relevant here (explaining a strange peculiarity of the language).  The C++ reference does the same with side notes mentioning future changes or how to disambiguate things that aren't obvious.

The information only needs to live in one place.  `if let` would be fine, with `while let` linking to `if let` for more information.  I might even trim `while let` even more to remove redundant information, but doesn't necessarily need to be done in this PR.

Reply:
Putting it in a note and deduplicating seems like a reasonable resolution to me.